<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>946</width>
    <height>645</height>
   </rect>
  </property>
  <property name="maximumSize">
   <size>
    <width>16777215</width>
    <height>16777215</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QFrame" name="frame">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>20</y>
      <width>421</width>
      <height>71</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true"/>
    </property>
    <property name="frameShape">
     <enum>QFrame::StyledPanel</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Raised</enum>
    </property>
    <widget class="QPushButton" name="Addfile">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>-10</y>
       <width>84</width>
       <height>71</height>
      </rect>
     </property>
     <property name="minimumSize">
      <size>
       <width>10</width>
       <height>10</height>
      </size>
     </property>
     <property name="toolTip">
      <string>Add DBC path</string>
     </property>
     <property name="styleSheet">
      <string notr="true"> QPushButton#Addfile {
     background-color: transparent;
     border:none
 }
 QPushButton#Addfile:pressed {
     background-color: rgb(220,220,220);     
	 border:none
 }
</string>
     </property>
     <property name="text">
      <string/>
     </property>
     <property name="icon">
      <iconset>
       <normaloff>toppng.png</normaloff>toppng.png</iconset>
     </property>
     <property name="iconSize">
      <size>
       <width>80</width>
       <height>80</height>
      </size>
     </property>
    </widget>
    <widget class="QLabel" name="label">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>50</y>
       <width>71</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>DBC path:</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="lineEdit">
     <property name="geometry">
      <rect>
       <x>72</x>
       <y>50</y>
       <width>161</width>
       <height>20</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">background-color: transparent;
border: none</string>
     </property>
    </widget>
   </widget>
   <widget class="Line" name="line">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>90</y>
      <width>921</width>
      <height>16</height>
     </rect>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="QFrame" name="frame_2">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>110</y>
      <width>921</width>
      <height>481</height>
     </rect>
    </property>
    <property name="frameShape">
     <enum>QFrame::StyledPanel</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Raised</enum>
    </property>
    <widget class="QPushButton" name="log">
     <property name="geometry">
      <rect>
       <x>120</x>
       <y>50</y>
       <width>101</width>
       <height>111</height>
      </rect>
     </property>
     <property name="toolTip">
      <string>load logging file</string>
     </property>
     <property name="styleSheet">
      <string notr="true"> QPushButton#log {
     background-color: transparent;
     border:none
 }
 QPushButton#log:pressed {
     background-color: rgb(220,220,220);     
     border:none
 }
</string>
     </property>
     <property name="text">
      <string/>
     </property>
     <property name="icon">
      <iconset>
       <normaloff>icons8-log-96.png</normaloff>icons8-log-96.png</iconset>
     </property>
     <property name="iconSize">
      <size>
       <width>70</width>
       <height>70</height>
      </size>
     </property>
    </widget>
    <widget class="Line" name="line_4">
     <property name="geometry">
      <rect>
       <x>340</x>
       <y>110</y>
       <width>20</width>
       <height>211</height>
      </rect>
     </property>
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
    </widget>
    <widget class="Line" name="line_7">
     <property name="geometry">
      <rect>
       <x>470</x>
       <y>350</y>
       <width>41</width>
       <height>41</height>
      </rect>
     </property>
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
    <widget class="Line" name="line_8">
     <property name="geometry">
      <rect>
       <x>460</x>
       <y>90</y>
       <width>20</width>
       <height>281</height>
      </rect>
     </property>
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
    </widget>
    <widget class="QPushButton" name="mode_button">
     <property name="geometry">
      <rect>
       <x>340</x>
       <y>140</y>
       <width>121</width>
       <height>161</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">background-color: transparent;
border:none

</string>
     </property>
     <property name="text">
      <string/>
     </property>
     <property name="icon">
      <iconset>
       <normaloff>offline.png</normaloff>
       <normalon>online.png</normalon>
       <selectedoff>offline.png</selectedoff>
       <selectedon>online.png</selectedon>offline.png</iconset>
     </property>
     <property name="iconSize">
      <size>
       <width>100</width>
       <height>100</height>
      </size>
     </property>
     <property name="checkable">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="Line" name="line_9">
     <property name="geometry">
      <rect>
       <x>220</x>
       <y>90</y>
       <width>131</width>
       <height>41</height>
      </rect>
     </property>
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
    <widget class="Line" name="line_2">
     <property name="geometry">
      <rect>
       <x>430</x>
       <y>210</y>
       <width>81</width>
       <height>31</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">color: black
</string>
     </property>
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
    <widget class="Line" name="line_5">
     <property name="geometry">
      <rect>
       <x>350</x>
       <y>210</y>
       <width>21</width>
       <height>31</height>
      </rect>
     </property>
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
    <widget class="QPushButton" name="log_path">
     <property name="geometry">
      <rect>
       <x>600</x>
       <y>330</y>
       <width>81</width>
       <height>81</height>
      </rect>
     </property>
     <property name="toolTip">
      <string>choose logging path</string>
     </property>
     <property name="whatsThis">
      <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;br/&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
     </property>
     <property name="styleSheet">
      <string notr="true"> QPushButton {
     background-color: transparent;
     border:none
 }
 QPushButton:pressed {
     background-color: rgb(220,220,220);
     border:none     
 }
</string>
     </property>
     <property name="text">
      <string/>
     </property>
     <property name="icon">
      <iconset>
       <normaloff>logging_file.png</normaloff>logging_file.png</iconset>
     </property>
     <property name="iconSize">
      <size>
       <width>80</width>
       <height>80</height>
      </size>
     </property>
    </widget>
    <widget class="QPushButton" name="print_data">
     <property name="geometry">
      <rect>
       <x>600</x>
       <y>50</y>
       <width>81</width>
       <height>81</height>
      </rect>
     </property>
     <property name="toolTip">
      <string>open message displayer</string>
     </property>
     <property name="styleSheet">
      <string notr="true"> QPushButton {
     background-color: transparent;
     border:none
 }
 QPushButton:pressed {
     background-color: rgb(220,220,220);     
 }
</string>
     </property>
     <property name="text">
      <string/>
     </property>
     <property name="icon">
      <iconset>
       <normaloff>TRACE.png</normaloff>TRACE.png</iconset>
     </property>
     <property name="iconSize">
      <size>
       <width>70</width>
       <height>70</height>
      </size>
     </property>
    </widget>
    <widget class="QPushButton" name="CAN_bus">
     <property name="geometry">
      <rect>
       <x>100</x>
       <y>240</y>
       <width>149</width>
       <height>70</height>
      </rect>
     </property>
     <property name="toolTip">
      <string>initial CAN bus</string>
     </property>
     <property name="styleSheet">
      <string notr="true"> QPushButton#CAN_bus {
     background-color: transparent;
     border:none
 }
 QPushButton#CAN_bus:pressed {
     background-color: rgb(220,220,220);     
     border:none
 }
</string>
     </property>
     <property name="text">
      <string/>
     </property>
     <property name="icon">
      <iconset>
       <normaloff>CAN.png</normaloff>CAN.png</iconset>
     </property>
     <property name="iconSize">
      <size>
       <width>70</width>
       <height>70</height>
      </size>
     </property>
    </widget>
    <widget class="QLineEdit" name="lineEdit_0">
     <property name="geometry">
      <rect>
       <x>110</x>
       <y>323</y>
       <width>149</width>
       <height>20</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">background-color: transparent;
border: none</string>
     </property>
     <property name="text">
      <string>CAN 0    500 kbaud</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="lineEdit_1">
     <property name="geometry">
      <rect>
       <x>110</x>
       <y>350</y>
       <width>149</width>
       <height>20</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">background-color: transparent;
border: none</string>
     </property>
     <property name="text">
      <string>CAN 1    500 kbaud</string>
     </property>
    </widget>
    <widget class="Line" name="line_10">
     <property name="geometry">
      <rect>
       <x>290</x>
       <y>300</y>
       <width>61</width>
       <height>41</height>
      </rect>
     </property>
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
    <widget class="QLabel" name="label_2">
     <property name="geometry">
      <rect>
       <x>94</x>
       <y>136</y>
       <width>91</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string>logging file:</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="lineEdit_2">
     <property name="geometry">
      <rect>
       <x>170</x>
       <y>140</y>
       <width>151</width>
       <height>16</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">background-color: transparent;
border: none</string>
     </property>
    </widget>
    <widget class="Line" name="line_11">
     <property name="geometry">
      <rect>
       <x>680</x>
       <y>350</y>
       <width>81</width>
       <height>41</height>
      </rect>
     </property>
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
    <widget class="QPushButton" name="sav_data">
     <property name="geometry">
      <rect>
       <x>770</x>
       <y>330</y>
       <width>81</width>
       <height>81</height>
      </rect>
     </property>
     <property name="toolTip">
      <string>save logging file</string>
     </property>
     <property name="whatsThis">
      <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;br/&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
     </property>
     <property name="styleSheet">
      <string notr="true"> QPushButton {
     background-color: transparent;
     border:none
 }
 QPushButton:pressed {
     background-color: rgb(220,220,220);     
     border:none
 }
</string>
     </property>
     <property name="text">
      <string/>
     </property>
     <property name="icon">
      <iconset>
       <normaloff>save.png</normaloff>save.png</iconset>
     </property>
     <property name="iconSize">
      <size>
       <width>60</width>
       <height>60</height>
      </size>
     </property>
    </widget>
    <widget class="QPushButton" name="add_bus_save">
     <property name="geometry">
      <rect>
       <x>500</x>
       <y>350</y>
       <width>51</width>
       <height>41</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">background-color: transparent;
border:none

</string>
     </property>
     <property name="text">
      <string/>
     </property>
     <property name="icon">
      <iconset>
       <normaloff>add_bus.png</normaloff>add_bus.png</iconset>
     </property>
     <property name="iconSize">
      <size>
       <width>20</width>
       <height>20</height>
      </size>
     </property>
     <property name="checkable">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="Line" name="line_12">
     <property name="geometry">
      <rect>
       <x>540</x>
       <y>350</y>
       <width>41</width>
       <height>41</height>
      </rect>
     </property>
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
    <widget class="QLineEdit" name="lineEdit_3">
     <property name="geometry">
      <rect>
       <x>590</x>
       <y>420</y>
       <width>141</width>
       <height>16</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">background-color: transparent;
border: none</string>
     </property>
    </widget>
    <widget class="QPushButton" name="add_bus_graphic">
     <property name="geometry">
      <rect>
       <x>500</x>
       <y>200</y>
       <width>51</width>
       <height>41</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">background-color: transparent;
border:none

</string>
     </property>
     <property name="text">
      <string/>
     </property>
     <property name="icon">
      <iconset>
       <normaloff>add_bus.png</normaloff>add_bus.png</iconset>
     </property>
     <property name="iconSize">
      <size>
       <width>20</width>
       <height>20</height>
      </size>
     </property>
     <property name="checkable">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="QPushButton" name="add_bus_trace">
     <property name="geometry">
      <rect>
       <x>500</x>
       <y>70</y>
       <width>51</width>
       <height>41</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">background-color: transparent;
border:none

</string>
     </property>
     <property name="text">
      <string/>
     </property>
     <property name="icon">
      <iconset>
       <normaloff>add_bus.png</normaloff>add_bus.png</iconset>
     </property>
     <property name="iconSize">
      <size>
       <width>20</width>
       <height>20</height>
      </size>
     </property>
     <property name="checkable">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="Line" name="line_16">
     <property name="geometry">
      <rect>
       <x>470</x>
       <y>70</y>
       <width>41</width>
       <height>41</height>
      </rect>
     </property>
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
    <widget class="Line" name="line_15">
     <property name="geometry">
      <rect>
       <x>540</x>
       <y>70</y>
       <width>41</width>
       <height>41</height>
      </rect>
     </property>
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
    <widget class="QPushButton" name="trace_graphic">
     <property name="geometry">
      <rect>
       <x>600</x>
       <y>180</y>
       <width>81</width>
       <height>81</height>
      </rect>
     </property>
     <property name="toolTip">
      <string>show graphics</string>
     </property>
     <property name="styleSheet">
      <string notr="true"> QPushButton {
     background-color: transparent;
     border:none
 }
 QPushButton:pressed {
     background-color: rgb(220,220,220);     
 }
</string>
     </property>
     <property name="text">
      <string/>
     </property>
     <property name="icon">
      <iconset>
       <normaloff>Graphic.png</normaloff>Graphic.png</iconset>
     </property>
     <property name="iconSize">
      <size>
       <width>70</width>
       <height>70</height>
      </size>
     </property>
    </widget>
    <widget class="Line" name="line_3">
     <property name="geometry">
      <rect>
       <x>540</x>
       <y>210</y>
       <width>41</width>
       <height>31</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">color: black
</string>
     </property>
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
   </widget>
   <widget class="QFrame" name="frame_3">
    <property name="geometry">
     <rect>
      <x>450</x>
      <y>20</y>
      <width>481</width>
      <height>71</height>
     </rect>
    </property>
    <property name="frameShape">
     <enum>QFrame::StyledPanel</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Raised</enum>
    </property>
    <widget class="QPushButton" name="Run_3">
     <property name="geometry">
      <rect>
       <x>420</x>
       <y>10</y>
       <width>44</width>
       <height>51</height>
      </rect>
     </property>
     <property name="minimumSize">
      <size>
       <width>10</width>
       <height>10</height>
      </size>
     </property>
     <property name="toolTip">
      <string>clean all</string>
     </property>
     <property name="styleSheet">
      <string notr="true"> QPushButton#Run_3 {
     background-color: transparent;
     border:none
 }
 QPushButton#Run_3:pressed {
     background-color: rgb(220,220,220);    
     border:none 
 }
</string>
     </property>
     <property name="text">
      <string/>
     </property>
     <property name="icon">
      <iconset>
       <normaloff>clean.png</normaloff>clean.png</iconset>
     </property>
     <property name="iconSize">
      <size>
       <width>50</width>
       <height>50</height>
      </size>
     </property>
    </widget>
    <widget class="QPushButton" name="Run_2">
     <property name="geometry">
      <rect>
       <x>360</x>
       <y>10</y>
       <width>44</width>
       <height>51</height>
      </rect>
     </property>
     <property name="minimumSize">
      <size>
       <width>10</width>
       <height>10</height>
      </size>
     </property>
     <property name="toolTip">
      <string>stop</string>
     </property>
     <property name="styleSheet">
      <string notr="true">QPushButton {
     background-color: transparent;
     border:none
 }
 QPushButton:pressed {
     background-color: rgb(220,220,220);  
     border:none 
 }
</string>
     </property>
     <property name="text">
      <string/>
     </property>
     <property name="icon">
      <iconset>
       <normaloff>pause-button.png</normaloff>pause-button.png</iconset>
     </property>
     <property name="iconSize">
      <size>
       <width>45</width>
       <height>45</height>
      </size>
     </property>
    </widget>
    <widget class="QPushButton" name="Run">
     <property name="geometry">
      <rect>
       <x>290</x>
       <y>10</y>
       <width>44</width>
       <height>51</height>
      </rect>
     </property>
     <property name="minimumSize">
      <size>
       <width>10</width>
       <height>10</height>
      </size>
     </property>
     <property name="toolTip">
      <string>run</string>
     </property>
     <property name="styleSheet">
      <string notr="true"> QPushButton#Run {
     background-color: transparent;
     border:none
 }
 QPushButton#Run:pressed {
     background-color: rgb(220,220,220);    
     border:none 
 }
</string>
     </property>
     <property name="text">
      <string/>
     </property>
     <property name="icon">
      <iconset>
       <normaloff>toppng(1).png</normaloff>toppng(1).png</iconset>
     </property>
     <property name="iconSize">
      <size>
       <width>40</width>
       <height>40</height>
      </size>
     </property>
    </widget>
   </widget>
  </widget>
 </widget>
 <resources/>
 <connections/>
</ui>
